frontend/src/pages/SetEdit/index.tsx

```diff
@@ -0,0 +1,123 @@
+import "./style.css";
+import { FlashCard } from "../../components/FlashCard.tsx";
+import { useState } from "preact/hooks";
+import { useLocation } from "preact-iso";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { getSetById } from "../../api/sets.ts";
+import { createCard } from "../../api/cards.ts";
+
+export default function SetEditPage() {
+  const location = useLocation();
+  const queryClient = useQueryClient();
+
+  // Extract set id from the path like /set/:id
+  // use window.location.pathname because the LocationHook doesn't expose pathname
+  const _g = globalThis as unknown as { location?: { pathname?: string } };
+  const pathname = _g.location?.pathname ?? "";
+  //regular expression to extract set id
+  const match = pathname.match(/\/set\/([^/]+)/);
+  const setId = match ? match[1] : "";
+
+  const {
+    data: set,
+    isLoading,
+    error,
+  } = useQuery({
+    queryKey: ["set", setId],
+    queryFn: () => getSetById(setId),
+    enabled: !!setId,
+  });
+
+  const [front, setFront] = useState("");
+  const [back, setBack] = useState("");
+
+  const createCardMutation = useMutation({
+    mutationFn: (params: { set_id: string; front: string; back: string }) =>
+      createCard(params),
+    onSuccess: () => {
+      // Refresh the set so new card appears
+      queryClient.invalidateQueries({ queryKey: ["set", setId] });
+      setFront("");
+      setBack("");
+    },
+  });
+
+  const handleCreate = (e: Event) => {
+    e.preventDefault();
+    if (!front.trim() || !back.trim()) {
+      alert("Both front and back must be provided");
+      return;
+    }
+    createCardMutation.mutate({
+      set_id: setId,
+      front: front.trim(),
+      back: back.trim(),
+    });
+  };
+
+  if (!setId) {
+    return <div class="set-edit-page">No set id provided in URL</div>;
+  }
```

Comment:
this should handle when the set is invalid, as the route should always have `:id`